### ISSUE TITLE
Support the -WhatIf switch on AzureResourceManagerTemplateDeploymentV3

### ISSUE THREAD

**Issue by nnellans** [microsoft/azure-pipelines-tasks#16110]
## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Feature

**Enter Task Name**:  AzureResourceManagerTemplateDeployment

## Environment
- Server - Azure Pipelines or TFS on-premises:  Azure Pipelines
    
- Agent - Hosted or Private:  Hosted, ubuntu-latest

## Issue Description

Would really love to use this task to run a deployment with the -WhatIf switch.  Would it be possible to add this as an option to the task?  Thanks.

**Comment by PiotrWachulec** [comment]
Yes, we are waiting for a long time for that feature! It's reeeeealy needed!

**Comment by Manbearpiet** [comment]
Would still like this too

**Comment by PiotrWachulec** [comment]
@Manbearpiet to be honest, we switched to PowerShell deployment. It's faster and we have more flexibility in the deployment configuration. I'm recommending this approach :)

**Comment by Manbearpiet** [comment]
> @Manbearpiet to be honest, we switched to PowerShell deployment. It's faster and we have more flexibility in the deployment configuration. I'm recommending this approach :)

Yeah we use CLI, but still would be a nice feature to have :-)

**Comment by geik** [comment]
Oh, yes please! ... I'd really like this feature.

**Comment by benjaminpieplow** [comment]
This would be a huge help for newcomers to Azure ARM/BICEP infrastructure management. Being able to easily gauge the result of a deployment, instead of "muck around and find out", eases a lot of the worry about breaking something, and the (less confident) people this would help the most likely don't have the coding skills to build PowerShell deployment wrappers for their projects. Having a `deploymentMode: 'what-if'` would be optimal.

**Comment by PolkageistHS** [comment]
2+ years and no response.  This would be very helpful for me as well.

**Comment by nnellanspdl** [comment]
2.5 years since opening this issue.  I'm going to say this is probably never happening ☹️

**Comment by denys-artiukhov** [comment]
Would be very helpful to have this feature!

**Comment by mischa69** [comment]
Please, with sugar on it :-)
Github azure/bicep-deploy@v1 already supports this function

**Comment by stijnherreman** [comment]
Duplicate issues closed by the stale bot:
* https://github.com/microsoft/azure-pipelines-tasks/issues/14690
* https://github.com/microsoft/azure-pipelines-tasks/issues/15514
* https://github.com/microsoft/azure-pipelines-tasks/issues/16837

**Comment by geik** [comment]
Any updates?  (Please?)

**Comment by aldass** [comment]
Bump... this is still needed!!!!

**Comment by nnellanspdl** [comment]
Not stale, still an issue.

Day number 1,347 of this issue being open.

**Comment by nnellanspdl** [comment]
There is now a BicepDeploy task, which supports what-if.

https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/BicepDeployV0
